src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -279,30 +278,21 @@ internal static unsafe byte[] UnsafeDecode(ReadOnlyMemory<char> str)
             }
             else
             {
-                if (decodedLength == str.Length)
+                if (decodedLength == strSpan.Length)
                 {
-                    if (MemoryMarshal.TryGetArray(str, out ArraySegment<char> segment))
-                    {
-                        return Convert.FromBase64CharArray(segment.Array, segment.Offset, segment.Count);
-                    }
-                    else
-                    {
-                        bool gotString = MemoryMarshal.TryGetString(str, out string text, out int start, out int length);
-                        Debug.Assert(gotString, "Expected ReadOnlyMemory to wrap either array or string");
-                        return Convert.FromBase64String(text.Substring(start, length));
-                    }
+                    return Convert.FromBase64CharArray(strSpan.ToArray(), 0, strSpan.Length);
```

Comment:
Converting the span into a char Array irrespective of whether the underlying structure is an array or a string. Thoughts on this approach?

Reply:
does create an additional allocation. Could you use `Encoding.GetBytes` method to convert the `ReadOnlySpan<char>` to a `byte[]`. This would avoid the overhead of allocating and copying to a new `char[]`
for example:
```csharp
byte[] bytes = Encoding.UTF8.GetBytes(strSpan);
return Convert.FromBase64CharArray(bytes, 0, bytes.Length);
```
but I'm not sure.

Reply:
why are we not using: https://learn.microsoft.com/en-us/dotnet/api/system.convert.tryfrombase64chars?view=net-8.0

Reply:
Convert.TryFromBase64Chars is not available on netstandard2.0, net472 and its predecessors

Reply:
can we profile to see the difference? if it is meaningful, it is worth the ifdef

Reply:
Convert.TryFromBase64Chars is already in use under a preprocessor directive [on Ln239](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blame/6bea7839b1f3c1dfb14502a304ecc14b4782bad2/src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs#L239)

Reply:
ah, you are correct. thanks